typed-protocols/src/Network/TypedProtocol/Codec.hs

```diff
@@ -242,6 +244,10 @@ runDecoderPure runM decoder bs = runM (runDecoder bs =<< decoder)
 data AnyMessage ps where
      AnyMessage :: Message ps st st' -> AnyMessage ps
 
+-- requires UndecidableInstances
+instance (forall st st'. Show (Message ps st st')) => Show (AnyMessage ps) where
```

Comment:
Ah and this one does require `QuantifiedConstraints`. Lets say that too.

As in the other module, add a comment where we introduce `UndecidableInstances` to remind to audit new instances defined here to make sure we're not creating problems for new instances. We can note we've checked two, the Show and the Excetpion one. Both are ok.